queries/c/highlights.scm

```diff
@@ -96,12 +101,13 @@
 
 (field_identifier) @property
 (statement_identifier) @label
-(type_identifier) @type
-(primitive_type) @type
-(sized_type_specifier) @type
 
-((identifier) @type
```

Comment:
Why remove this? This is the only chance how we can detect types when their are used (I know this convention is often only used for C++). What else would a upper case identifier mean (even in C)?

C could use additionally `foo_t` with regex "_t$" for types.

If you think this is to C++-like, move this to C++-highlights?

Reply:
No not at all but that actually creates wrong highlights, please refer to my other comment at the same point.

Reply:
This version : `(binary_expression left: (identifier) @type)` actually does highlight the types too in declarations, and also avoid some highlighting errors.
I think that I have misses the same highlight in the `cast` nodes though.

Reply:
What expression would cause a binary_expression for example?
Using the naming convention is better (query is copied from C++ upstream) since we can not cover all situations where a type can occur. In C and especially C++ a type can occur in virutally any identifier position.

Reply:
> in declarations

As I said in my previous comment

> In C and especially C++ a type can occur in virtually any identifier position.

I agree on that, but anyway, I can find examples in nvim codebase (actually the screenshot...) where this way of highlighting type is actually wrong, which, in my opinion is a problem. I can revert that, but I think that we will want a better way of highlighting types.

Reply:
Maybe you're using a old parser version of the C parser. I can not achieve any highlight with your query. Can you give an example expression?

Well I am not arguing that you should remove your more sophisticated expression, but as a naming convention can serve as a basic guess (they are a bit more applied for C++ code. So moving this to C++ would be an option)

Reply:
> Maybe you're using a old parser version of the C parser. I can not achieve any highlight with your query. Can you give an example expression? I tried various types of declarations.

That's weird... It worked for me and outputed the screenshot, for example `TSNode *node = foo();` should work.

Reply:
![image](https://user-images.githubusercontent.com/7189118/86778691-f1877f00-c05a-11ea-8c23-2a6b64490c5d.png)

Reply:
TSInstall C ?

Reply:
> TSInstall C ?

Hehe I am not that dumb ;-)

I might have missed something, in any case, this way should work and is better in my opinion than relying on naming conventions:
```
(declaration type: (_) @type)
(cast_expression type: (_) @type)
```

We have a syntax tree in our hands, relying too much on naming conventions is the job of regex based syntax highlighting.

Reply:
Maybe the built-in C parser of Nvim is old.

Reply:
https://medium.com/@mujjingun_23509/full-proof-that-c-grammar-is-undecidable-34e22dd8b664

Reply:
```scheme
(declaration type: (_) @type)
(cast_expression type: (_) @type)
```
that's not necessary a type as a whole. In C++ you can have an expression that evaluates to a type or you have `foo::Xer::Type_T` where only the last part is the type. So `(declaration type: (identifier) @type)` would be ok. C++ relies on naming conventions to decide whether the last part of a namespace experssion is a type `foo::Xer::Type_T`

Reply:
But hey, that's C++ not C 😉
Don't mix things up, the goal here is to fix C highlights, I understand that you use it for C++, and I can add that very same line in the C++ highlights query, but actually, C does not need something this general.

Reply:
> Maybe the built-in C parser of Nvim is old.

I think that this was a mistake on my side.

Reply:
> But hey, that's C++ not C wink
> Don't mix things up, the goal here is to fix C highlights, I understand that you use it for C++, and I can add that very same line in the C++ highlights query, but actually, C does not need something this general.

This is why I am suggesting to move that query to the C++ highlights.

Reply:
Should be all good for now, I use something that will for sure be fine for C but not for C++, the `binary_expression` thing was actually a parsing error (see the comment below) I will open an issue rn to report that.